service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -461,8 +456,10 @@ public ResponseEntity<ApiCreatedControlledGcpAiNotebookInstanceResult> createAiN
     Workspace workspace =
         workspaceService.validateWorkspaceAndAction(
             userRequest, workspaceUuid, getSamAction(body.getCommon()));
+    // AI Notebooks are zonal resources. So, don't parse the region from the zone.
     String resourceLocation =
-        getResourceLocation(workspace, body.getAiNotebookInstance().getLocation());
+        gcpCloudContextService.getResourceLocation(
```

Comment:
Is "location" not specific enough to capture region and zone?
Maybe name the call `getZonedRegion()` or some such.